pkg/trace/api/otlp.go

```diff
@@ -94,6 +115,23 @@ func (o *OTLPReceiver) Stop() {
 // Export implements ptraceotlp.Server
 func (o *OTLPReceiver) Export(ctx context.Context, in ptraceotlp.ExportRequest) (ptraceotlp.ExportResponse, error) {
 	defer timing.Since("datadog.trace_agent.otlp.process_grpc_request_ms", time.Now())
+
+	select {
+	// Wait for the semaphore to become available, allowing the handler to
+	// decode its payload.
+	// Afer the configured timeout, respond without ingesting the payload,
+	// returning an error.
+	case o.recvsem <- struct{}{}:
```

Comment:
Rather than having this pattern here, and in `pkg/trace/api/api.go`, and in every other possible future sender -- why not have it on the [receiving side](https://github.com/DataDog/datadog-agent/blob/1e82b81b3a4398d9663baaf106d0aa90c36e6ee3/pkg/trace/agent/agent.go#L187-L193)? I may be missing something but I wanted to ask the question.

Reply:
Yeah, it may be nicer to wrap this in a type or make it part of the `*Agent` or something. I'll take a look.